src/biome/text/_model.py

```diff
@@ -307,12 +309,48 @@ def predict_batch(
         try:
             predictions = self.forward_on_instances(instances)
         except Exception as error:
-            raise WrongValueError(f"Failed to make predictions for '{input_dicts}'") from error
+            raise WrongValueError(
+                f"Failed to make predictions for '{input_dicts}'"
+            ) from error
         for input_dict, prediction in zip(input_dicts, predictions):
             self.log_prediction(input_dict, prediction)
 
         return predictions
 
+    def explain_batch(self, input_dicts: Iterable[Dict[str, Any]], n_steps: int):
+        """
+        Applies a batch prediction including token attribution explanation
+
+        Parameters
+        ----------
+
+        input_dicts
+            The input data. The keys of the dicts must comply with the `self.inputs` attribute
+        n_steps: int
+            The number of steps for token attribution calculation (if proceed).
+            If the number of steps is less than 1, the attributions will not be calculated
+
+        Returns
+        -------
+            The predictions with information of internal data representation.
+
+        """
+        instances = [self.text_to_instance(**input_dict) for input_dict in input_dicts]
+
+        try:
+            predictions = self.forward_on_instances(instances)
+        except Exception as error:
+            raise WrongValueError(
+                f"Failed to make predictions for '{input_dicts}'"
+            ) from error
+
+        explained_predictions = [
+            self.__build_explained_prediction(prediction, instance, n_steps)
+            for instance, prediction in zip(instances, predictions)
+        ]
```

Comment:
I think this loop is not necessary. In the explain methods of the heads, we always embed the [instance in a batch](https://github.com/recognai/biome-text/blob/master/src/biome/text/modules/heads/classification/text_classification.py#L84). So the input to the `IntegratedGradients.attribute` is already a batch.
But let's leave this possible optimization to another PR (Maybe add an TODO here).

Reply:
We will tackle in another PR. We need provide a common mechanism for prediction --> explain conversion if we accept explain prediction batches.